### ISSUE TITLE
`Incorrect array size` error for some cases of reversed ranges with absolute addressing (previously: Partially locked, reversed range breaks setCellContents)

### ISSUE THREAD

**Issue by AJNandi** [handsontable/hyperformula#1106]
### Description

setCellContents on hyperformula instance with the following data causes an "Incorrect array size" error. 

```
const tableData = [
  [1, "=SUM(A1:A$1)"],
  [1, "=SUM(A2:A$1)"]
];
```

### Demo
https://codesandbox.io/s/frosty-meadow-umnbk2?file=/src/data.js

Partially locked ranges that are in the correct order work.
Calculating a single reversed and locked formula seems to work fine. See this repo here: https://codesandbox.io/s/silly-leftpad-d38lgu?file=/src/data.js

HyperFormula version: 2.2

### Links
Related #835

**Comment by sequba** [comment]
Hi @AJNandi, thanks for your report. It seems to be a very peculiar bug, but - obviously - this is not the intended behavior of HyperFormula so we'll address it in one of the upcoming releases. I hope it's not a blocker for you ;)

**Comment by sequba** [comment]
@AJNandi,
 
I'm more than happy to announce that we just released [HyperFormula 2.3.1](https://github.com/handsontable/hyperformula/releases/tag/2.3.1) where this issue is fixed. We are closing this issue as solved. If there is anything that won't work for you after updating, please leave a comment.
